### PR TITLE
fix parameter name and order in FindTransparencyValues

### PR DESCRIPTION
**Pull request by bcko**
- related to #2348 , #2554
- addresses https://github.com/diasurgical/devilutionX/pull/2554#discussion_r685753643

### CODE REVIEW

**Comment by ephphatha** [comment]
You've pretty much got the code to the point that all the `FloodTransparancyValues()` functions can be combined and take floorId as a parameter to that function :). That'd make this code significantly less repetitive if you felt like making that change too.

**Comment by bcko** [comment]
> You've pretty much got the code to the point that all the `FloodTransparancyValues()` functions can be combined and take floorId as a parameter to that function :). That'd make this code significantly less repetitive if you felt like making that change too.

Sounds great! I will update the pull request to combine `FloodTransparancyValues()`. Thank you @ephphatha !

**Comment by bcko** [comment]
Updated the pull request to refactor FloodTransparencyValues().

File: Source/gendung.h [hunk 1]
```diff
@@ -248,5 +248,6 @@ void DRLG_Init_Globals();
 bool SkipThemeRoom(int x, int y);
 void InitLevels();
 void FindTransparencyValues(int i, int j, int x, int y, int d, uint8_t floorID);
```
Comment (ephphatha) [inline Source/gendung.h:250]: This is no longer called from outside gendung.h so no longer needs to be in the header.

File: Source/drlg_l1.cpp [hunk 2]
```diff
@@ -2201,6 +2184,7 @@ void GenerateLevel(lvl_entry entry)
 	}
 
 	bool doneflag;
+	uint8_t floorID = 13;
```
Comment (AJenbo) [inline Source/drlg_l1.cpp:2187]: Not much point in having a local variable for this when it just gets pass to another function

File: Source/gendung.cpp [hunk 3]
```diff
@@ -597,4 +597,21 @@ void FindTransparencyValues(int i, int j, int x, int y, int d, uint8_t floorID)
 	FindTransparencyValues(i + 1, j + 1, x + 2, y + 2, 8, floorID);
 }
```
Comment (ephphatha) [inline Source/gendung.cpp:598]: `FindTransparencyValues` can be moved into the anonymous namespace at the start of gendung.cpp now that it's only called from one version of `FloodTransparencyValues` :D

File: Source/drlg_l4.cpp [hunk 4]
```diff
@@ -1323,6 +1306,7 @@ void GenerateLevel(lvl_entry entry)
 	constexpr int Minarea = 173;
 	int ar;
 	bool doneflag;
+	uint8_t floorID = 6;
```
Comment (ephphatha) [inline Source/drlg_l4.cpp]:
```suggestion
	constexpr uint8_t HellFloorID = 6;
```
This variable is not modified so can be a constexpr, also suggested a more descriptive name to differentiate the different uses of the resulting function.

File: Source/drlg_l2.cpp [hunk 5]
```diff
@@ -2921,6 +2904,7 @@ void FixDoors()
 void GenerateLevel(lvl_entry entry)
 {
 	bool doneflag = false;
+	uint8_t floorID = 3;
```
Comment (ephphatha) [inline Source/drlg_l2.cpp]:
```suggestion
	constexpr uint8_t CatacombsFloorID = 3;
```
Same situation as the previous comment :)

**Review by AJenbo — APPROVED** [review]

File: Source/drlg_l1.cpp (hunk 2 above)
Comment (bcko) [reply Source/drlg_l1.cpp]: Updated the pull request to remove local variable floorID in all three places. Thank you @AJenbo !

Comment (ephphatha) [inline Source/drlg_l1.cpp]:
```suggestion
	constexpr uint8_t CatacombsFloorID = 13;
```
Last one. Recent changes for globals/constants have been moving to PascalCase as well, so trying to keep that convention where possible.

**Review by ephphatha — COMMENTED** [review]
Looks good, there's a couple of cleanups made possible now that you've removed the duplicates :D

File: Source/drlg_l1.cpp (hunk 2 above)
Comment (ephphatha) [reply Source/drlg_l1.cpp]: I was too slow 😅 . You can ignore that part of my comments.

File: Source/drlg_l4.cpp (hunk 4 above)
Comment (AJenbo) [reply Source/drlg_l4.cpp]: I think this is used in multiple places so maybe it belongs in a more general cleanup

**Comment by bcko** [comment]
@ephphatha updated the pull request to address `FindTransparencyValues()` in `gendung.h` and `gendung.cpp`. Thank you!

**Review by AJenbo — APPROVED** [review]

**Review by ephphatha — APPROVED** [review]

**Comment by AJenbo** [comment]
@bcko looks like you need to login to CircleCi for it to be able to run the tests on your PRs

**Comment by bcko** [comment]
> @bcko looks like you need to login to CircleCi for it to be able to run the tests on your PRs

Thank you for letting me know! @AJenbo I just logged into CircleCI

image

**Pull request merged**